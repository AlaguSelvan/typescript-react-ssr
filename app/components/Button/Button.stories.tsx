import React from 'react'
import Button from './Button'

export default {
  component: Button,
  title: 'Button'
}

export const Base = () => {
  // const [title, updateTitle] = useState('');

  // https://github.com/WordPress/gutenberg/pull/18031
  // useEffect(() => {
  // updateTitle(document.title);
  // }, []);

  return (
    <>
      <Button />
      {/* <p>title: {title}</p> */}
    </>
  )
}
