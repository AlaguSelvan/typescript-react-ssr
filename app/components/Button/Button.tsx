import React from 'react';
import { css } from 'emotion';

const color = 'white';

const Button = (props: any) => {
  const [count, setCount] = React.useState(0);
  return (
    <button
      onClick={() => setCount(count + 2)}
      className={css`
        padding: 32px;
        background-color: red;
        font-size: 24px;
        border-radius: 4px;
        &:hover {
          color: ${color};
        }
      `}
    >
      {props.children} {count} times
    </button>
  );
};

export default Button;
