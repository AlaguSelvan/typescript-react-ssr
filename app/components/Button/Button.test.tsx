import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Button from './Button';

afterEach(cleanup);

test('should match snapshot', () => {
  const { asFragment } = render(<Button />);
  expect(asFragment).toMatchSnapshot();
});

test('has text when text prop is passed', () => {
  const value = 'hello';
  const { getByTestId } = render(<Button value="hello" />);
  expect(getByTestId('para')).toHaveTextContent(value);
});
