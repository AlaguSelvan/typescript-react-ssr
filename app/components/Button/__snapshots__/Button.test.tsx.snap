// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a message 1`] = `[Function]`;

exports[`should match snapshot 1`] = `[Function]`;
